conn.go

```diff
@@ -182,14 +196,29 @@ func (conn *Conn) Close() error {
 		conn.outLck.Unlock()
 		return nil
 	}
+
 	close(conn.out)
 	conn.closed = true
 	conn.outLck.Unlock()
+
+	close(conn.shutdownSigWorker)
+	conn.signalQueueLck.Lock()
+	// This is to avoid unnecessary wakeup of sigWorker
+	tmp := conn.signalQueueWakeUp
```

Comment:
I understand that you can treat conn.signalQueueWakeUp as a pointer to channel. So tmp holds a pointer to channel and conn.signalQueueWakeUp is nil. So from this place sigWorker can call select on either nilled conn.signalQueueWakeUp (which is blocking forever) or on closed conn.shutdownSigWorker (which returns immediately).